src/Aspire.Hosting.AWS/CloudFormation/CloudFormationExtensions.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon.CloudFormation;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS;
+using Aspire.Hosting.AWS.CloudFormation;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting;
+
+public static class CloudFormationExtensions
+{
+    /// <summary>
+    /// Add a CloudFormation stack for provisioning application resources.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="stackName">The name of the CloudFormation stack.</param>
+    /// <param name="templatePath">The path to the CloudFormation template that defines the CloudFormation stack.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ICloudFormationResource> AddAWSCloudFormationProvisioning(this IDistributedApplicationBuilder builder, string stackName, string templatePath)
+    {
+        var resource = new CloudFormationResource(stackName, templatePath);
+        var cfBuilder = builder.AddResource(resource);
+
+        builder.Services.AddLifecycleHook<CloudFormationLifeCycle>(sp => ActivatorUtilities.CreateInstance<CloudFormationLifeCycle>(sp, resource));
+        return cfBuilder;
+    }
+
+    /// <summary>
+    /// The AWS credential profile to use for resolving credentials to make AWS service API calls.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="awsSdkConfig">The name of the AWS credential profile.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithAWSSDKReference(this IResourceBuilder<ICloudFormationResource> builder, IResourceBuilder<IAWSSDKConfigResource> awsSdkConfig)
+    {
+        builder.Resource.AWSSDKConfig = awsSdkConfig.Resource;
+        return builder;
+    }
+
+    /// <summary>
+    /// The configured Amazon CloudFormation service client used to make service calls.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="cloudFormationClient">The AWS CloudFormation service client.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithAWSCloudFormationClient(this IResourceBuilder<ICloudFormationResource> builder, IAmazonCloudFormation cloudFormationClient)
```

Comment:
This feels like it should come from DI, not via an argument on a method.

Reply:
I updated the docs on this method. My thoughts was giving a user the ability to override the normal DI or `IAWSSDKConfig` behavior. For example somebody that has done "extra" configuration to have the CloudFormation service client point to a LocalStack instance. It isn't strictly necessary now and could remove for now.